pandas/tests/base/test_misc.py

```diff
@@ -147,6 +147,11 @@ def test_searchsorted(request, index_or_series_obj):
     # See gh-12238
     obj = index_or_series_obj
 
+    if any(isinstance(x, str) for x in obj) and any(isinstance(x, int) for x in obj):
```

Comment:
Can you just do `if obj.inferred_type == "mixed"`? And everywhere below where you need to xfail this new index